csharp/src/Google.Protobuf.Test.NativeAot/Program.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Google.Protobuf.Test.NativeAot.Reflection;
+
+try
```

Comment:
I think it would be good to include a descriptive comment, either here or in a README.md, to explain the purpose of all of this code, and the reason for not using a regular test framework. (I'm pretty sure I get it, but clarity's good.)